S3Scanner/__main__.py

```diff
@@ -137,6 +138,8 @@ def main():
     parser_group.add_argument('--buckets-file', '-f', dest='buckets_file',
                               help='Name of text file containing bucket names to check', metavar='file')
     parser_group.add_argument('--bucket', '-b', dest='bucket', help='Name of bucket to check', metavar='bucket')
+    parser_scan.add_argument('--json', '-j', dest='json_file', help='Name of file to write JSON to (defaults to `results.json`)',
```

Comment:
Let's also add a unit test for this new functionality. Tests should verify:

- Scenario: the JSON file doesn't already exist. 
  - Scan a bucket
  - Assert the file gets created
  - Assert the contents are expected
- Scenario: The JSON file already exists and has content
  - Scan a bucket
  - Assert the contents of the file are what was there plus the bucket we just scanned